valohai_yaml/objs/input.py

```diff
@@ -37,13 +44,15 @@ def __init__(
         description: Optional[str] = None,
         keep_directories: KeepDirectoriesValue = False,
         filename: Optional[str] = None,
+        override_mode: Optional[InputOverrideMode] = InputOverrideMode.APPEND,
     ) -> None:
         self.name = name
         self.default = default  # may be None, a string or a list of strings
         self.optional = bool(optional)
         self.description = description
         self.keep_directories = KeepDirectories.cast(keep_directories)
         self.filename = filename
+        self.override_mode = override_mode
```

Comment:
This should probably do `OverrideMode.cast(override_mode)` like `keep_directories` above (so you'd need to implement `OverrideMode.cast` too).